enterprise/app/api_keys/api_keys.tsx

```diff
@@ -235,6 +236,15 @@ export default class ApiKeysComponent extends React.Component<ApiKeysComponentPr
     this.onChangeCapability(request, api_key.ApiKey.Capability.CACHE_WRITE_CAPABILITY, !e.target.checked, onChange);
   }
 
+  private onChangeVisibility<T extends ApiKeyFields>(
+    request: T,
+    onChange: (name: string, value: any) => any,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    request.visibleToDevelopers = e.target.checked;
+    onChange("visibleToDevelopers", request.visibleToDevelopers);
```

Comment:
```suggestion
    onChange("visibleToDevelopers", e.target.checked);
```

(`onChange` winds up calling `setState` with a new request where the provided field is equal to the value passed in)